Source/Core/InputCommon/DynamicInputTextureConfiguration.cpp

```diff
@@ -0,0 +1,261 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "InputCommon/DynamicInputTextureConfiguration.h"
+
+// Ignore opencv warnings
+#pragma warning(push, 0)
+#include <opencv2/core/core.hpp>
+#include <opencv2/features2d/features2d.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/imgproc/types_c.h>
+#include <opencv2/opencv.hpp>
+#pragma warning(pop)
+#include <picojson/picojson.h>
+
+#include <fstream>
+
+#include "Common/FileUtil.h"
+#include "Common/StringUtil.h"
+#include "Core/ConfigManager.h"
+#include "InputCommon/ControllerEmu/ControllerEmu.h"
+
+#include "VideoCommon/HiresTextures.h"
+#include "VideoCommon/TextureCacheBase.h"
+
+namespace
+{
+cv::Mat resizeKeepAspectRatio(const cv::Mat& input, const cv::Size& dstSize,
+                              const cv::Scalar& bgcolor)
+{
+  cv::Mat output;
+
+  double h1 = dstSize.width * (input.rows / (double)input.cols);
+  double w2 = dstSize.height * (input.cols / (double)input.rows);
+  // initially no borders
+  int top = 0;
+  int down = 0;
+  int left = 0;
+  int right = 0;
+  if (h1 <= dstSize.height)
+  {
+    // only vertical borders
+    auto diff = dstSize.height - h1;
+    if (diff == 1)
+    {
+      top = 1;
+      down = 0;
+    }
+    else
+    {
+      top = diff / 2;
+      down = top;
+    }
+    cv::resize(input, output, cv::Size(dstSize.width, h1));
+  }
+  else
+  {
+    // only horizontal borders
+    auto diff = dstSize.width - w2;
+    if (diff == 1)
+    {
+      left = 1;
+      right = 0;
+    }
+    else
+    {
+      left = diff / 2;
+      right = left;
+    }
+    cv::resize(input, output, cv::Size(w2, dstSize.height));
+  }
+
+  cv::copyMakeBorder(output, output, top, down, left, right, cv::BORDER_CONSTANT, bgcolor);
+
+  return output;
+}
+}  // namespace
+
+namespace InputCommon
+{
+DynamicInputTextureConfiguration::DynamicInputTextureConfiguration(const std::string& json_file)
+{
+  std::ifstream json_stream(json_file);
```

Comment:
You need to use `OpenFStream` within `Common/FileUtil.h`, otherwise this can fail to open on systems with different locales.